checker.cpp

```diff
@@ -1,17 +1,122 @@
 #include <assert.h>
+#include<iostream>
+using namespace std;
 
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
-    return false;
-  } else if(respRate < 30 || respRate > 95) {
-    return false;
-  }
-  return true;
-}
+class Alert
+{
+    public:
+    virtual void raiseAlert(const char* vitalName,const char* level)=0;  //It tell that
+
+};
+class AlertWithSMS: public Alert
+{
+public:
+    void raiseAlert(const char* vitalName,const char* level)
+    {
+        cout<<"SMS: "<<vitalName<<" "<<level<<endl;
+    }
+};
+
+class AlertWithSound: public Alert
+{
+public:
+    void raiseAlert(const char* vitalName,const char* level)
+    {
+        cout<<"Sound: "<<vitalName<<" "<<level<<endl;
+    }
+
+};
+
+class RangeChecker
+{
+private:
+    int lower,upper;
+    const char* vitalName;
+    Alert* alerter;
+public:
+    RangeChecker(const char* name,int low,int up,Alert* alerterPtr)
+    {
+        vitalName=name;
+        lower=low;
+        upper=up;
+        alerter=alerterPtr;
+    }
+    void checkAgainstRange(float value)
+    {
+        if(value < lower)
+        {
+
+            alerter->raiseAlert(vitalName,"Too Low");
+        }
+        else if(value > upper)
+        {
+
+            alerter->raiseAlert(vitalName,"Too High");
+        }
+    }
+
+};
+
+struct vitalParameter
+{
+    float bpm, spo2, respRate, ecg, bodyTemp, glucose,
```

Comment:
Nicely structured solution and extra parameters added 👍 
Can you think of a way to express that a few parameters have not been included? Like glucose not being monitored for a patient who has no diabetes?